### ISSUE TITLE
flux-accounting service: need a way to configure permissions for `flux account` commands

### ISSUE THREAD

**Issue by cmoussa1** [flux-framework/flux-accounting#323]
In an offline conversation, there was a need brought up to be able to set permissions on the available list of commands in flux-accounting. Currently, the systemd unit file for flux-accounting is configured to the following:

```
User=flux
Group=flux
```

Which means the flux-accounting service is run under user `flux`. If you are a normal user and you try to run a flux-accounting command, you are met with an error message saying you don't have credentials. This is because you are sending a message to the flux-accounting service that carries your UID:

```
PermissionError: [Errno 1] Request requires owner credentials
```

There will be a need to configure the commands (or the RPC that is sent to the service as a result of calling the command?) to only be able to be run by certain users or groups. For example, a normal user on a cluster should have permissions to read from the database (i.e, view their own user information with `flux account view-user`), but should not be able to add other users or edit their own information. Administrators on a cluster, however, should be able to have access to the entire suite of flux-accounting commands: adding, removing, editing user information, etc. 

One suggestion to start would be to send read-only RPCs for a regular user who just wants to be able to read information from the DB.

**Comment by cmoussa1** [comment]
Per our team meeting yesterday, @ryanday36 let us know that it is okay for users to see each other's information from the flux-accounting DB, which contains information like shares, fair share, job usage, etc. So this might make adding read-only RPCs a little simpler for at least the `view-*` commands, which only read from the DB. There are also commands that we would probably want to keep _out_ of just anybody's hands, like adding new users, banks, and editing information. These commands might require some sort of verification that the user trying to issue the RPC has the permissions to do so.

**Comment by ryanday36** [comment]
This sounds about right. I'm good with all users being able to run the `view-` commands. I'd also love to be able to give a more limited set of users (e.g. members of the `lcstaff` unix group) `add-`, `edit-`, and `delete-` access, but I can probably also do that with `sudo` if necessary.

**Comment by garlick** [comment]
It should be pretty easy to open a given RPC up to all users.  The default is for the handler to be only available to the instance owner.  To open it up, you have to set a `rolemask` to FLUX_ROLE_USER when the handler is registered.

Then all users will be able to send a request to the message handler.  What you do with it then is up to you.   You have access to the userid who is making the request so you _could_ add logic at that point to do something like limit them to only their own info.  It sounds like that's not needed here though.

Of course you could use that technique to allow a configured list of users to access the "admin" level accounting commands, but I wouldn't recommed it.  Long term, we should consider adding a "role" for this class of users, and then instead of setting the rolemask of some admin command to USER, we could just  set it to the new role.  That requires there to be some additional work in flux-core that we haven't gotten to yet, but it is consistent with the design.  Short term, I would suggest creative `sudoers` editing as @ryanday36 suggested.  Note that the command only needs to run as the `flux` user.  Root also works but isn't required.

**Comment by cmoussa1** [comment]
Thanks @garlick, that's really helpful! I had a couple of clarifying questions if that's okay:

> The default is for the handler to be only available to the instance owner. To open it up, you have to set a rolemask to FLUX_ROLE_USER when the handler is registered.
>
> Then all users will be able to send a request to the message handler.

This sounds good. I think I'm not quite clear on where this `rolemask` is set up. Do you set it up when you call `msg_watcher_create()`?

```python
self.handle.msg_watcher_create(
    view_user, FLUX_MSGTYPE_REQUEST, "accounting.view-user", self
).start()
```

Sorry, I've probably missed it somewhere.

**Comment by grondo** [comment]
Ah, there may not be a way to add a rolemask to a msg_watcher from Python. We'll have to fix that first.

**Comment by grondo** [comment]
Actually this might work for now (but it is not ideal)  if you want to start testing:
```python
watcher = self.handle.msg_watcher_create(
    view_user, FLUX_MSGTYPE_REQUEST, "accounting.view-user", self
)
self.handle.msg_handler_allow_rolemask(watcher.handle, FLUX_ROLE_ALL)
watcher.start()
```

**Comment by cmoussa1** [comment]
Thanks, I'll try this out and report back!

**Comment by cmoussa1** [comment]
This should be fixed by #330 now, so I'll close this. We can always re-open a more specific issue if we find something related to this issue.
